packages/common/src/stores/ProfileController.ts

```diff
@@ -27,45 +27,48 @@ const profileSchema = yup.object().shape({
 @injectable()
 export default class ProfileController {
   private readonly profileService?: ProfileService;
+  private readonly storageService: StorageService;
 
-  constructor(@inject('INTEGRATION_TYPE') integrationType: IntegrationType) {
+  constructor(@inject(INTEGRATION_TYPE) integrationType: IntegrationType, storageService: StorageService) {
     this.profileService = getNamedModule(ProfileService, integrationType, false);
+    this.storageService = storageService;
   }
 
-  private getSandbox = () => {
-    return useConfigStore.getState().isSandbox ?? true;
+  private isValidProfile = (profile: unknown): profile is ProfilesData => {
+    try {
+      profileSchema.validateSync(profile);
+      return true;
+    } catch (e: unknown) {
+      return false;
+    }
   };
 
+  isEnabled() {
+    return !!this.profileService;
+  }
+
   listProfiles = async () => {
     assertModuleMethod(this.profileService?.listProfiles, 'listProfiles is not available in profile service');
 
-    const res = await this.profileService.listProfiles(undefined, this.getSandbox());
-
-    const { canManageProfiles } = useProfileStore.getState();
-
-    if (!canManageProfiles && res?.responseData.canManageProfiles) {
-      useProfileStore.setState({ canManageProfiles: true });
-    }
```

Comment:
This one is tricky. There can be JWP account service with profiles enabled, however request to the profiles endpoint could return 403 or 400 meaning that the feature is not enabled.

Reply:
This was redundant because the `useProfiles` hook exposes the `canManageProfiles` if the endpoint returns an error.

Reply:
What if we we open the home page, go to the account page and then back to the home page? Won't profiles be fetched several times in this case?